src/differentials.jl

```diff
@@ -206,3 +206,27 @@ end
 end
 
 Base.conj(x::Thunk) = @thunk(conj(extern(x)))
+
+
+####
+"""
+    NO_FIELDS
+
+Constant for the reverse-mode derivative with respect to structure that has no fields.
+The most notable use for this is for the reverse-mode derivative with respect to the
+function itself, when that function is not a closure.
+"""
+const NO_FIELDS = DNE()
+
+####
+"""
+    differential(𝒟::Type, der)
+
+For some differential (e.g. a `Number`, `AbstractDifferential`, `Matrix` etc),
+converts it to another differential that is more suited for the domain given by
```

Comment:
```suggestion
convert it to another differential that is more suited for the domain given by
```
 Imperative form sounds better to me here.